opensrp-anc/src/main/java/org/smartregister/anc/library/listener/ContactTaskDisplayClickListener.java

```diff
@@ -0,0 +1,227 @@
+import org.smartregister.anc.library.model.Task;
+import org.smartregister.anc.library.util.ConstantsUtils;
+import org.smartregister.anc.library.util.ContactJsonFormUtils;
+import org.smartregister.util.FormUtils;
+
+import java.util.Calendar;
+
+import timber.log.Timber;
+
+public class ContactTaskDisplayClickListener implements View.OnClickListener {
+    private ProfileTasksFragment profileTasksFragment;
+    private FormUtils formUtils;
+
+    public ContactTaskDisplayClickListener(ProfileTasksFragment profileTasksFragment) {
+        this.profileTasksFragment = profileTasksFragment;
+    }
+
+    @Override
+    public void onClick(View view) {
+        if (view != null) {
+            if (view.getId() == R.id.accordion_info_icon) {
+                infoAlertDialog(view);
+            } else if (view.getId() == R.id.undo_button) {
+                undoTasksEntries(view);
+            } else {
+                prepFormForLaunch(view);
+            }
+        }
+    }
+
+    /**
+     * This performs all the necessary calculations to get the form ready for launch. This updates the title,
+     * Adds the new form fields
+     *
+     * @param view {@link View}
+     */
+    private void prepFormForLaunch(View view) {
+        Context context = ((Context) view.getTag(R.id.accordion_context));
+        Task task = ((Task) view.getTag(R.id.task_object));
+        JSONObject taskValue = ((JSONObject) view.getTag(R.id.accordion_jsonObject));
+
+        if (context != null && task != null && taskValue != null) {
+            JSONArray subFormFields = loadSubFormFields(taskValue, context);
+            String formTitle = getFormTitle(taskValue);
+            JSONObject form = loadTasksForm(context);
+            updateFormTitle(form, formTitle);
+            updateFormFields(form, subFormFields);
+
+            profileTasksFragment.startTaskForm(form);
+        }
+    }
+
+    /**
+     * Updates the form step1 title to match the test header
+     *
+     * @param form  {@link JSONObject}
+     * @param title {@link String}
+     */
+    private void updateFormTitle(JSONObject form, String title) {
+        try {
+            if (form != null && StringUtils.isNotBlank(title)) {
+                if (form.has(JsonFormConstants.STEP1)) {
+                    JSONObject stepOne = form.getJSONObject(JsonFormConstants.STEP1);
+                    stepOne.put(JsonFormConstants.STEP_TITLE, title);
+                }
+            }
+        } catch (JSONException e) {
+            Timber.e(e, " --> updateFormTitle");
+        }
+    }
+
+    /**
+     * Add the sub form fields to the main form for loading.
+     *
+     * @param form   {@link JSONObject}
+     * @param fields {@link JSONArray}
+     */
+    private void updateFormFields(JSONObject form, JSONArray fields) {
+        try {
+            if (form != null && fields != null && fields.length() > 0) {
+                if (form.has(JsonFormConstants.STEP1)) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/manual/OpenSRP/opensrp-client-anc/pullRequest?prid=4809639)